kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/service/impl/KutsuServiceImpl.java

```diff
@@ -52,25 +52,27 @@
     @Override
     @Transactional
     public long createKutsu(KutsuCreateDto dto) {
-         if (!kutsuRepository.listKutsuListDtos(new KutsuCriteria().withTila(AVOIN).withSahkoposti(dto.getSahkoposti())).isEmpty()) {
+         if (!kutsuRepository.listKutsuListDtos(new KutsuCriteria().withTila(AVOIN).withSahkoposti(dto.getSahkoposti()),
+                 KutsuOrganisaatioOrder.AIKALEIMA.getSortWithDirection()).isEmpty()) {
              throw new IllegalArgumentException("kutsu_with_sahkoposti_already_sent");
          }
-        
-        Kutsu entity = mapper.map(dto, Kutsu.class);
 
-        entity.setId(null);
-        entity.setAikaleima(LocalDateTime.now());
-        entity.setKutsuja(getCurrentUserOid());
-        entity.setSalaisuus(UUID.randomUUID().toString());
-        entity.setTila(AVOIN);
+        final Kutsu newKutsu = mapper.map(dto, Kutsu.class);
+
+        newKutsu.setId(null);
+        newKutsu.setAikaleima(LocalDateTime.now());
+        newKutsu.setKutsuja(getCurrentUserOid());
+        newKutsu.setSalaisuus(UUID.randomUUID().toString());
+        newKutsu.setTila(AVOIN);
+        newKutsu.getOrganisaatiot().forEach(kutsuOrganisaatio -> kutsuOrganisaatio.setKutsu(newKutsu));
 
-        validator.validate(entity);
+        validator.validate(newKutsu);
 
-        entity = kutsuRepository.persist(entity);
+        this.kutsuDataRepository.save(newKutsu);
```

Comment:
https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#save-S-

`Use the returned instance for further operations as the save operation might have changed the entity instance completely.` Eli `newKutsu = this.kutsuDataRepository.save(newKutsu);`

Reply:
Totta. Täytyy tehdä uusi muuttuja kun tuota käytetään lambdassa jolloin sen pitäisi olla final.